fido/fido.py

```diff
@@ -91,6 +91,13 @@ def finished_errorback(error):
     bodyProducer = None
     if body:
         bodyProducer = FileBodyProducer(StringIO(body))
+        # content-length needs to be removed because it was computed based on
+        # body but body is now being processed by twisted FileBodyProducer
+        # causing content-length to lose meaning and break the client.
+        # FileBodyProducer will take care of re-computing length and re-adding
+        # a new content-length header later.
+        headers.pop('Content-Length', None)
```

Comment:
this is modifying the input headers, I would copy the headers if we need to modify them